cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/selector/StorageSelectorConfig.java

```diff
@@ -0,0 +1,49 @@
+package com.linkedin.openhouse.cluster.storage.selector;
+
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configures the StorageSelector bean for storage-selector configured in {@link StorageProperties}
+ * The return value of the bean is the {@link StorageSelector} implementation that matches the name
+ * in storage-selector or is null if the storage selector is not configured.
+ */
+@Slf4j
+@Configuration
+public class StorageSelectorConfig {
+
+  @Autowired StorageProperties storageProperties;
+
+  @Autowired List<StorageSelector> storageSelectors;
+
+  /**
+   * Checks the name of storage-selector from {@link StorageProperties} against all implementations
+   * of {@link StorageSelector} and returns the implementation that matches the name. returns null
+   * if not configured
+   *
+   * @return
+   */
+  @Bean("StorageSelector")
+  StorageSelector provideStorageSelector() {
+    String selectorName;
```

Comment:
can you add a check to see if storage selector is not provided in yaml, if so, select `DefaultStorageSelector`

Reply:
done

Reply:
can we add a precondition based approach,  the current code might catch irrelevant NPEs which should be surfaced.
for example:
```
provideStorageSelector
{
Bool isStorageSelectorConfigured = /**/
if (not isStorageSelectorConfigured) {
  return DefaultStorageSelector
}
else {
  /* select storage*/
  /* if storageName not found throw illegalArgumentException saying Unkown selector: NAME  configured */
}

}
```

Reply:
addressed. Please check